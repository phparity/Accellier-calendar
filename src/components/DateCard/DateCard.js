import React from "react";
import PropTypes from 'prop-types';
import "./DateCard.scss";

/**
 * Renders a date card component with the given DateText and DayText.
 *
 * @param {string} DateText - The date to be displayed.
 * @param {string} DayText - The day of the week to be displayed.
 * @param {string} className - Additional CSS class for styling.
 * @returns {JSX.Element} The rendered date card component.
 */
function DateCard({ DateText, DayText, className }) {
  return (
    <div data-testid="date-card-outer" className={`date_card_otr ${className}`}>
      <div className="date_card_inr">
        <h3 className="date_text">{DateText}</h3>
        <p className="day_text">{DayText}</p>
      </div>
    </div>
  );
}

// Define prop types for the component
DateCard.propTypes = {
  DateText: PropTypes.string,
  DayText: PropTypes.string,
  className: PropTypes.string
};

export default DateCard;
