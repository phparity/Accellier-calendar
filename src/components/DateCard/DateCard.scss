@import '../../assets/common-style/variable.scss';

.date_cards_otr {
    display: flex;
    align-items: stretch;
    .date_card_otr {
        width: 100%;
        &:not(:last-child){
            border-right: 1px solid $Border-color; 
        } 
    }
    .date_card_inr {
        display: flex;
        align-items: center;
        justify-content: center;
        flex-direction: column;
        padding: 0 10px;
        height: 68px;
        gap: 6px;
        cursor: pointer;
        transition: .3s;
        -webkit-transition: .3s;
        -moz-transition: .3s;
        -ms-transition: .3s;
        -o-transition: .3s; 

        @media (max-width:1700px){
            height: 71px;
            gap: 5px; 
        } 
        &:hover {
            background-color: $Light-Bg3;
        }
    }
    .date_text {
        font-size: 16px !important;
        line-height: 20px;
        font-family: $font-family-Normal;  
        color: $Black;
        margin: 0;   
    }
    .day_text {
        font-size: 12px;
        font-weight: 400;  
        line-height: 15px;
        color: $Text-color;
        font-family: $font-family-Normal;
        margin: 0; 
    }
}

.current_date {
    .date_card_inr {
        background-color: $Light-Bg3;
    }
}