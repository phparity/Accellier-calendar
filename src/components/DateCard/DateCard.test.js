import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import DateCard from "./DateCard";

describe("DateCard Component", () => {
  it("renders DateText and DayText correctly", () => {
    const dateText = "2024-02-08";
    const dayText = "Monday";

    const { getByText } = render(
      <DateCard DateText={dateText} DayText={dayText} />,
    );

    const renderedDateText = getByText(dateText);
    const renderedDayText = getByText(dayText);

    expect(renderedDateText).toBeInTheDocument();
    expect(renderedDayText).toBeInTheDocument();
  });

  it("applies custom className correctly", () => {
    const customClassName = "custom-class";
    const { container } = render(
      <DateCard
        DateText="2024-02-08"
        DayText="Monday"
        className={customClassName}
      />,
    );

    const dateCardElement = container.querySelector(
      ".date_card_otr.custom-class",
    );

    expect(dateCardElement).toBeInTheDocument();
  });

  it("renders correctly", () => {
    const dateText = "2024-02-08";
    const dayText = "Monday";
    const tree = renderer
      .create(<DateCard DateText={dateText} DayText={dayText} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
