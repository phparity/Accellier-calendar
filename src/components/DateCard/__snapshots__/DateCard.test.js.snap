// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateCard Component renders correctly 1`] = `
<div
  className="date_card_otr undefined"
  data-testid="date-card-outer"
>
  <div
    className="date_card_inr"
  >
    <h3
      className="date_text"
    >
      2024-02-08
    </h3>
    <p
      className="day_text"
    >
      Monday
    </p>
  </div>
</div>
`;
